src/geofabrics/dem.py

```diff
@@ -1383,20 +1453,29 @@ def add_lidar(
                 )
             )
             buffered_foreshore = buffered_foreshore.overlay(
-                self.catchment_geometry.full_land, how="difference", keep_geom_type=True
+                self.catchment_geometry.full_land,
+                how="difference",
+                keep_geom_type=True,
             )
             # Clip DEM to buffered foreshore
-            mask = numpy.logical_not(
-                numpy.isnan(
-                    dem.z.rio.clip(buffered_foreshore.geometry, drop=False).data
-                )
+            foreshore_mask = (
+                dem.z.rio.clip(buffered_foreshore.geometry, drop=False).notnull().data
             )
 
-            # get coarse DEM points on the foreshore - with any positive set to zero
-            dem.z.data[mask & (dem.z.data > 0)] = 0
+            # Set any positive LiDAR foreshore points to zero
+            dem.data_source.data = numpy.where(
+                foreshore_mask & (dem.z.data > 0),
+                self.SOURCE_CLASSIFICATION["ocean bathymetry"],
+                dem.data_source.data,
+            )
+            dem.lidar_source.data = numpy.where(
+                foreshore_mask & (dem.z.data > 0),
+                self.SOURCE_CLASSIFICATION["no data"],
+                dem.lidar_source.data,
+            )
+            dem.z.data = numpy.where(foreshore_mask & (dem.z.data > 0), 0, dem.z.data)
```

Comment:
Same as before, I would suggest, when possible, to use the xarray API when possible instead of using numpy (which might also trigger unwanted computations if the operation is not supported by dask.array... I suspect this is not the case here :)).

Reply:
done